v4/export/ir_impl_test.go

```diff
@@ -27,43 +24,54 @@ func (s *simpleRowReceiver) BindAddress(args []interface{}) {
 	}
 }
 
-func (s *testIRImplSuite) TestRowIter(c *C) {
+func TestRowIter(t *testing.T) {
+	t.Parallel()
+
 	db, mock, err := sqlmock.New()
-	c.Assert(err, IsNil)
-	defer db.Close()
+	require.NoError(t, err)
+	defer func() {
+		_ = db.Close()
+	}()
 
 	expectedRows := mock.NewRows([]string{"id"}).
 		AddRow("1").
 		AddRow("2").
 		AddRow("3")
 	mock.ExpectQuery("SELECT id from t").WillReturnRows(expectedRows)
 	rows, err := db.Query("SELECT id from t")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	iter := newRowIter(rows, 1)
 	for i := 0; i < 100; i++ {
-		c.Assert(iter.HasNext(), IsTrue)
+		require.True(t, iter.HasNext())
 	}
+
 	res := newSimpleRowReceiver(1)
-	c.Assert(iter.Decode(res), IsNil)
-	c.Assert(res.data, DeepEquals, []string{"1"})
+	require.NoError(t, iter.Decode(res))
+	require.Equal(t, []string{"1"}, res.data)
+
 	iter.Next()
-	c.Assert(iter.HasNext(), IsTrue)
-	c.Assert(iter.HasNext(), IsTrue)
-	c.Assert(iter.Decode(res), IsNil)
-	c.Assert(res.data, DeepEquals, []string{"2"})
+	require.True(t, iter.HasNext())
```

Comment:
there were two `HasNext()` calls originally

```suggestion
	require.True(t, iter.HasNext())
	require.True(t, iter.HasNext())
```

Reply:
@kennytm I guess it is redundant occasionally? `iter.HasNext()` doesn't has side effect.

Reply:
I guess the point is to prove that HasNext has no side effect 🤔

Reply:
OK. I'll keep as is and postpone to any further comer.